### PR TITLE
init hoarctl grant seal/unseal

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <greg.hill@monax.io>

Tidy up hoarctl and add two CLI actions for sealing / unsealing a grant.

### CODE REVIEW

File: cmd/hoarctl/main.go [hunk 1]
```diff
@@ -92,93 +96,86 @@ func main() {
...
+		"Encrypt data from STDIN and return its reference.",
 		func(cmd *cli.Cmd) {
 
 			saltString := saltOpt(cmd)
 
 			cmd.Action = func() {
 				data, err := ioutil.ReadAll(os.Stdin)
 				if err != nil {
-					fatalf("Could read bytes from STDIN to store: %v", err)
+					fatalf("could not read bytes from STDIN to store: %v", err)
```
Comment (silasdavis) [inline cmd/hoarctl/main.go]: I know Go doesn't like capitalised errors... but this isn't an error and it won't be nested

File: cmd/hoarctl/main.go [hunk 2]
```diff
@@ -92,93 +96,86 @@ func main() {
...
-					fatalf("Error retrieving data: %v", err)
+					fatalf("error retrieving data: %v", err)
 				}
 				os.Stdout.Write(plaintext.Data)
 			}
 		})
 
 	hoarctlApp.Command("ref",
-		"Encrypt data from STDIN and return its reference",
+		"Encrypt data from STDIN and return its reference.",
```
Comment (silasdavis) [inline cmd/hoarctl/main.go]: Weeell I don't think we need a full stop here. See `docker`

File: grant/grant.go [hunk 3]
```diff
@@ -14,15 +14,15 @@ func Seal(secret SecretProvider, ref *reference.Ref, spec *Spec) (*Grant, error)
 	grt := &Grant{Spec: spec}
 
 	switch s := spec.GetValue().(type) {
-	case PlaintextSpec:
+	case *PlaintextSpec:
```
Comment (silasdavis) [inline grant/grant.go]: oh was this borked?

File: protobuf/grant.proto [hunk 4]
```diff
@@ -33,6 +33,6 @@ message SymmetricSpec {
 
 
 message OpenPGPSpec {
-    // TODO: figurate how to access local GPG entities, presuambly id
+    // TODO: figure out how to access local GPG entities, presumably id
```
Comment (silasdavis) [inline protobuf/grant.proto]: OMG the greglinter. What's wrong with 'figurate'

File: grant/grant.go (hunk 3 above)
Comment (gregdhill) [reply grant/grant.go]: Yeah I spent far too long trying to figure out why this catch wasn't working :P

File: protobuf/grant.proto (hunk 4 above)
Comment (gregdhill) [reply protobuf/grant.proto]: ITS NOT A WORD

File: cmd/hoarctl/main.go (hunk 1 above)
Comment (silasdavis) [reply cmd/hoarctl/main.go]: oh, though I was missing a not

File: grant/grant.go (hunk 3 above)
Comment (silasdavis) [reply grant/grant.go]: lol yeah that should have had a test but I didn't want you to not have any fun

File: cmd/hoarctl/main.go (hunk 1 above)
Comment (gregdhill) [reply cmd/hoarctl/main.go]: This is an error?

File: cmd/hoarctl/main.go (hunk 2 above)
Comment (gregdhill) [reply cmd/hoarctl/main.go]: Sure :+1:

**Comment by gregdhill** [comment]
@silasdavis all client methods are divided into separate files now :)

File: .goreleaser.yml [hunk 5]
```diff
@@ -13,8 +13,8 @@ builds:
     - -s -w
     - -X github.com/monax/hoard/project.commit={{.Commit}}
     - -X github.com/monax/hoard/project.date={{.Date}}
-- main: ./cmd/hoarctl
-  binary: hoarctl
+- main: ./cmd/hoctl
+  binary: hoctl
```
Comment (silasdavis) [inline .goreleaser.yml]: this .... looks weird to me, I think `hctl` is better

File: cmd/hoctl/local.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
+package main
```
Comment (silasdavis) [inline cmd/hoctl/local.go]: I can see where you are coming from, but I think this file should be named `encryption.go`. Other ops are 'local' but aren't in here, like `Reseal`. If we use the same name as the service who's methods are callable from here then we keep things consistent.

**Pull request merged**